pandas/core/arrays/integer.py

```diff
@@ -643,25 +665,30 @@ def cmp_method(self, other):
                 if len(self) != len(other):
                     raise ValueError("Lengths must match to compare")
 
-            # numpy will show a DeprecationWarning on invalid elementwise
-            # comparisons, this will raise in the future
```

Comment:
See previous question about this. Is this comment no longer relevant or correct? Or why was it removed?

Reply:
I'm not sure, do you know how this is actually hit? If NumPy is going to raise in the future, shouldn't they be seeing that warning?

Reply:
I think this is about the warning you get with comparisons with objects / non-broadcastable arrays. Eg:

```
In [29]: np.array([1, 2]) == "b"   
/home/joris/miniconda3/envs/dev/bin/ipython:1: FutureWarning: elementwise comparison failed; returning scalar instead, but in the future will perform elementwise comparison
  #!/home/joris/miniconda3/envs/dev/bin/python
Out[29]: False

In [30]: pd.array([1, 2]) == "b" 
Out[30]: array([False, False])
```

(it seems IntegerArray already handles this fine, not sure there is a explicit test for that)

Reply:
> it seems IntegerArray already handles this fine,

Gotch. It's silencing the same warning from NumPy, and falling back to `invalid_comparison`, which returns the expected result. I'll restore the comment.

Reply:
Actually... the comment is incorrect. NumPy will perform elementwise comparison in the future, not raise. If they were to raise on that in the future the implementation would be incorrect.

Though I'm still a bit confused, as the NumPy op is returning NotImplemented since we're calling it directly. Will that continue to return NotImplemented? Or will the elementwise result be different?